File: spec/javascript/test_str_encodings.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('StrEncodings', 'src/str_encodings.bin', function(r) {
    assert.equal("Some ASCII", r.str1);
    assert.equal("こんにちは", r.str2);
    assert.equal("こんにちは", r.str3);
    assert.equal("░▒▓", r.str4);
});

File: spec/lua/test_str_encodings.lua
local luaunit = require("luaunit")

require("str_encodings")

TestStrEncodings = {}

function TestStrEncodings:test_str_encodings()
    local r = StrEncodings:from_file("src/str_encodings.bin")

    luaunit.assertEquals(r.str1, "Some ASCII")
    luaunit.assertEquals(r.str2, "こんにちは")
    luaunit.assertEquals(r.str3, "こんにちは")
    luaunit.assertEquals(r.str4, "░▒▓")
end
